test-common/src/main/java/org/triplea/test/common/assertions/Optionals.java

```diff
@@ -0,0 +1,32 @@
+package org.triplea.test.common.assertions;
+
+import java.util.Optional;
+
+import org.hamcrest.Matcher;
+import org.triplea.test.common.CustomMatcher;
+
+/**
+ * Utility class with hamcrest matchers for {@code java.util.Optional}.
+ */
+public final class Optionals {
```

Comment:
Following Hamcrest conventions, this class should probably be named `OptionalMatchers`.

Reply:
This class is really similar to `Assertions.java`. I hope that Assertions.java will one day be broken up so it is topic specific. I like the idea of calling that class, `Throwables`, but not necessarily `ThrowableMatcher`.  To some extent this class is meant to mirror Assertions.java as it is similar; it was a choice whether to append more matchers to that class or create a new one. I've seen some good success in the simple naming convention to append `s` to the type being matched. It's very similar to `Collectors`, or `Collections`. 

@ssoloff I do recall seeing some matchers in classes suffixes with `Matchers`, but there are many that are not. Where is that hamcrest convention established?

There are not a lot of references for best practices of how to name such a class; I'm curious if you can find any better. Of some relevance I found: https://www.baeldung.com/hamcrest-custom-matchers. In that example the classes containing customer matchers have no suffix, but also similar for many matchers within hamrest, such as for example [IsCollectionWithSize](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/collection/IsCollectionWithSize.html)

Reply:
Our `Assertions` class contains assertion methods, similar to the JUnit `Assertions` class.  It contains no Hamcrest matchers, and thus I wouldn't expect it to have a `Matchers` suffix.

`Optionals` contains Hamcrest matchers for use _with_ an assertion method, typically `assertThat()`.

Yes, `Matcher` classes within Hamcrest typically do not have the `Matcher` suffix.  However, the _collections_ of matchers, usually used to group them in a single import, typically do have a `Matchers` suffix.  One usually does not import a Hamcrest matcher directly but rather uses a static method that exposes it from one of the matcher collection classes (e.g. `Matchers` or `CoreMatchers`).  (Unless, of course, it's such a rarely used matcher that no one ever bothered to add it one of the collection classes.)

Mockito also follows this convention for their own brand of (non-Hamcrest) matchers, e.g. `ArgumentMatchers`.

Reply:
This all may be a moot point now after 0fcf77df0.  It looks like `Optionals` is no longer used.

Reply:
> However, the collections of matchers, usually used to group them in a single import, typically do have a Matchers suffix. 

Can you give more examples? I can only find just a couple hamcrest examples following that convention. Looking at the bulk of hamcrest matchers, most do *not* follow that convention, and most of these classes do contains many matchers:  http://hamcrest.org/JavaHamcrest/javadoc/1.3/

I suspect the "Matchers" suffix is a mockito convention, not hamcrest.

Reply:
Sorry to pile on, but one thing I'm considering:
- imagine a class with a single hamcrest matcher, calling it "matchers" would be odd. It would also then be very odd for that class to need a rename just because we add a method to it. The other element to this, it seems like excessive details to wonder if it'll be "Matcher" or "matchers" depending on the public API count.

Reply:
> Can you give more examples?

The two matcher libraries I referenced in https://github.com/triplea-game/triplea/pull/4152#discussion_r223190962 both name their classes `OptionalMatchers`.

In addition, the other Spotify matchers also have a `Matchers` suffix.

> I would like to stick with the simple plural convention so we can be more similar to the core java library conventions, eg: Collectors, or Collections.

Those are more appropriate as extension-like methods for the targeted type (e.g. `Optionals` would contain utility or extension methods for `Optional`, similar to our existing `OptionalUtils` class).  Whereas, `OptionalMatchers` is effectively a container for one or more matchers that target `Optional`.

> imagine a class with a single hamcrest matcher, calling it "matchers" would be odd. It would also then be very odd for that class to need a rename just because we add a method to it. The other element to this, it seems like excessive details to wonder if it'll be "Matcher" or "matchers" depending on the public API count.

The matcher implementation classes would have a `Matcher` suffix.  The class with a `Matchers` suffix simply provides methods that return instances of the `*Matcher` classes.

Reply:
You're talking about classes that _implement_ `Matcher`.  I'm talking about classes that _provide_ matcher instances.

Yes, Hamcrest `Matcher` implementations typically do not include the `Matcher` suffix.  Probably because a) the whole library is focused on providing matchers, so they probably considered it redundant, and b) it is uncommon for clients to import individual matcher classes; they rather import the matcher container classes, such as `Matchers` and `CoreMatchers`.

My point about the Spotify library is that _all_ the matcher _container_ classes (which is what `Optionals` in this PR is) end with `Matchers`.  I agree that the convention for naming _individual_ matcher classes is not consistent.